### PR TITLE
Tag styling changes

### PR DESCRIPTION
**Pull request by veegandhi**
Tag component changes
## Description

Make svg optional for action button.
Introducing base class which allows to override baseClass of tag component. This will also allow to move away from accent property to completely custom style the tag as needed.
## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots (if appropriate):
**Default behaviour without svg**
image

**Default behaviour with baseClass and no svg**
image

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [x] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: docs/examples/TagExample.jsx [hunk 1]
```diff
@@ -106,6 +107,10 @@ const exampleProps = {
           propType: 'accent',
           type: 'string',
         },
+        {
+          propType: 'baseClass',
+          type: 'string',
```
Comment (ChaoLiangSuper) [inline docs/examples/TagExample.jsx]: should have the default value here.

File: src/components/alexandria/Tag/index.jsx [hunk 2]
```diff
@@ -4,22 +4,24 @@ import SvgSymbol from 'alexandria/SvgSymbol';
...
 import './styles.scss';
 
-const componentClass = 'tag-component';
+const defaultComponentClass = 'tag-component';
 
 export const ActionButton = ({ onAction, id, actionIconSvgHref }) => (
-  <span className="action-button" onClick={() => onAction(id)}>
-    <SvgSymbol href={actionIconSvgHref} />
+  <span className={`action-button${actionIconSvgHref ? '' : '-icon'}`} onClick={() => onAction(id)}>
+    {actionIconSvgHref ? <SvgSymbol href={actionIconSvgHref} /> : <span className="action-icon">x</span>}
```
Comment (ChaoLiangSuper) [inline src/components/alexandria/Tag/index.jsx]: The `x` should be `&#x2715;`

Comment (nimishjha) [inline src/components/alexandria/Tag/index.jsx]: I think the x should be a SVG or unicode symbol.

**Comment by ChaoLiangSuper** [comment]
Also after the discussing offline, I think the default style should change from 
screen shot 2018-11-21 at 2 53 19 pm
to
screen shot 2018-11-21 at 2 53 06 pm

(For a coloured tag)

Maybe let UX team to have a look.

File: docs/examples/TagExample.jsx (hunk 1 above)
Comment (veegandhi) [reply docs/examples/TagExample.jsx]: done

File: src/components/alexandria/Tag/index.spec.jsx [hunk 3]
```diff
@@ -70,4 +70,20 @@ describe('Tag', () => {
     );
     expect(component.find('.child')).to.have.length(2);
   });
+
+  it('should render an tag with base class', () => {
```
Comment (nimishjha) [inline src/components/alexandria/Tag/index.spec.jsx]: a tag? an <something> tag?

File: src/components/alexandria/Tag/styles.scss [hunk 4]
```diff
@@ -50,19 +51,18 @@
     fill: $color-inverse;
     fill-opacity: .6;
     height: $tag-height;
-    padding-top: $action-icon-padding;
     text-align: center;
     width: $tag-height;
+    padding-top: $action-icon-padding;
```
Comment (ChaoLiangSuper) [inline src/components/alexandria/Tag/styles.scss]: Instead setting the padding, how about using `flex` and `align-items: center`

File: src/components/alexandria/Tag/index.spec.jsx (hunk 3 above)
Comment (veegandhi) [reply src/components/alexandria/Tag/index.spec.jsx]: 😭 Whoops!

File: src/components/alexandria/Tag/index.jsx [hunk 5]
```diff
@@ -29,13 +31,16 @@ const Tag = ({ children, inverse, id, onAction, accent, actionIconSvgHref }) =>
   }
 
   if (onAction) {
-    classSuffixes.push('actionable');
+    defaultSuffixes.push('actionable');
```
Comment (ChaoLiangSuper) [inline src/components/alexandria/Tag/index.jsx]: Here instead `push` maybe we can call the `className` to handle the classnames

Comment (veegandhi) [reply src/components/alexandria/Tag/index.jsx]: It still needs to be prefixed with ```tag-component-```

File: src/components/alexandria/Tag/index.jsx [hunk 6]
```diff
@@ -29,13 +31,16 @@ const Tag = ({ children, inverse, id, onAction, accent, actionIconSvgHref }) =>
...
   }
 
-  const classes = classSuffixHelper({ classSuffixes, componentClass });
+  const classes = classSuffixHelper({ classSuffixes, componentClass: baseClass });
+  const defaultClasses = `${
```
Comment (xiaofan2406) [inline src/components/alexandria/Tag/index.jsx]: use `classnames` package

Comment (veegandhi) [reply src/components/alexandria/Tag/index.jsx]: https://github.com/Adslot/adslot-ui/pull/776#discussion_r235255014

File: src/components/alexandria/Tag/index.jsx (hunk 5 above)
Comment (xiaofan2406) [reply src/components/alexandria/Tag/index.jsx]: don't think classnames stop you from prefixing

**Pull request merged**